src/sys/linux_macos/dummy.rs

```diff
@@ -0,0 +1,40 @@
+//! Dummy FFI definitions needed for testing on non-Linux, non-OSX platforms
+use libc::{c_char, c_int, c_void, size_t, ssize_t};
+
+extern {
```

Comment:
Instead of specifying these with extern, mind implementing them with `unimplemented!()`. That's *slightly* safer.

Reply:
If I'm going to do that, then I should probably do the same for the BSD functions, when built on Linux or OSX, right?

Reply:
Ah, I see the issue. Actually... How about just changing the "is linux" check to "is not macos" and explain why? The entire *module* is usually cfged away so it shouldn't be an issue. That should be the simplest fix.

Reply:
Yes, that would be simpler.  But it wouldn't be any safer.  The other change I'd like to make would be to move all of the FFI definitions into libc.  That would require xattr to use its own `unimplemented` definitions for unsupported platforms.